src/t_set.c

```diff
@@ -948,6 +957,11 @@ void sinterGenericCommand(client *c, robj **setkeys,
         if (j == setnum) {
             if (cardinality_only) {
                 cardinality++;
+
+                /* We stop the searching after reaching the limit. */
+                if (limit && cardinality >= limit) {
+                    break;
+                }
```

Comment:
```suggestion
                if (limit && cardinality >= limit)
                    break;
```
Save one line of code.